### DISCUSSION TITLE
Execute _at a certain_ simulation time

### DISCUSSION THREAD

**Discussion by GiuseppeDiGuglielmo** [cocotb/cocotb#3718] [Q&A]
With cocotb we can wait for a certain amount of time or an event:

```
await Timer(10, units='ns')
cocotb.log.info("10 ns have passed")
await Timer(10, units='ns')
cocotb.log.info("10 ns have passed")
await Timer(10, units='ns')
cocotb.log.info("10 ns have passed")
```
or
```
await RisingEdge(rst)
cocotb.log.info("reset time, do something!")
```

But is it possible to wait and do something at a particular absolute simulation time? Something like in a coroutine _"wait for the simulation time 1024 ns (from time 0 ns) and then do something"_?

**Comment by marlonjames** [comment]
You can use [`cocotb.simulator.get_sim_time()`](https://docs.cocotb.org/en/latest/library_reference.html#cocotb.simulator.get_sim_time) to get the current time, and then use a `Timer` trigger to wait until your desired time step.
